diff --git a/node_modules/react-native-swipeable-flat-list/src/components/SwipeableFlatList.js b/node_modules/react-native-swipeable-flat-list/src/components/SwipeableFlatList.js
index 99f2d26..7ba1aff 100644
--- a/node_modules/react-native-swipeable-flat-list/src/components/SwipeableFlatList.js
+++ b/node_modules/react-native-swipeable-flat-list/src/components/SwipeableFlatList.js
@@ -23,11 +23,14 @@ export default class SwipeableFlatList extends Component {
 	constructor(props) {
 		super(props);
 		this._itemRef = React.createRef();
+		this.child = []
 	}
 
 	_handleOpenChild = (ref) => {
+		this._reset();
 		this._itemRef?.current?.close?.();
 		this._itemRef = ref;
+		this.child.push(ref)
 	}
 
 	_handleCloseChild = () => {
@@ -40,6 +43,11 @@ export default class SwipeableFlatList extends Component {
 		onScroll?.(event);
 	}
 
+	_reset = () => {
+		for (let i of this.child)
+			i.closeRight()
+	}
+
 	render = () => {
 		const {
 			data,
